File: scripts/algorithms/C/Consecutive Numbers Sum/Consecutive Numbers Sum.js
// Runtime: 101 ms (Top 40.00%) | Memory: 43.2 MB (Top 25.45%)

var consecutiveNumbersSum = function(n) {
    let count = 1;
    for (let numberOfTerms = 2; numberOfTerms < Math.sqrt(2*n) + 1; numberOfTerms++) {
        let startNumber = (n - numberOfTerms * (numberOfTerms - 1) / 2) / numberOfTerms;
        if (Number.isInteger(startNumber) && startNumber !== 0) {
            count++;
        }
    }
    return count;
};

File: scripts/algorithms/C/Consecutive Numbers Sum/Consecutive Numbers Sum.cpp
// Runtime: 6 ms (Top 49.82%) | Memory: 5.8 MB (Top 96.45%)
class Solution {
public:
    int consecutiveNumbersSum(int n) {
        int count = 0;
        for(int i = 2 ; i < n ; i++){
            int sum_1 = i*(i+1)/2;
            if(sum_1 > n)
                break;
            if((n-sum_1)%i == 0)
                count++;
        }
        return count+1;
    }
};